app/controllers/int_api/marketplaces_controller.rb

```diff
@@ -51,7 +51,7 @@ def create
 
     #include port separately to url_params if it was present
     if marketplace[:url] && marketplace[:url].split(':')[2]
-      url_params.merge!(port: marketplace[:url].split(':')[2])
+      url_params[:port] = marketplace[:url].split(':')[2]
```

Comment:
@ovan Thanks! This was a good clarification! 
I didn't have the difference between value type objects and composite data type objects so clear in mind, so indeed there was some misunderstanding that I'm more aware with now. 👍